assets/src/components/conversations/ConversationNotificationManager.ts

```diff
@@ -0,0 +1,157 @@
+import {Channel, Socket} from 'phoenix';
+import {throttle} from 'lodash';
+
+import logger from '../../logger';
+import {Conversation, Message} from '../../types';
+import {SOCKET_URL} from '../../socket';
+import * as API from '../../api';
+import {PhoenixPresence, PresenceDiff} from '../../presence';
+
+type Config = {
+  accountId: string;
+  onNewMessage: (message: Message) => void;
+  onNewConversation: (id: string) => void;
+  onConversationUpdated: (id: string, updates: Partial<Conversation>) => void;
+  onPresenceInit: (presence: PhoenixPresence) => void;
+  onPresenceDiff: (diff: PresenceDiff) => void;
+};
+
+/**
+ * ConversationNotificationManager manages the socket and channel used for
+ * connecting the React app with the Phoenix server for real-time messaging.
+ */
+class ConversationNotificationManager {
+  config: Config;
+  socket: Socket;
+  channel: Channel | null = null;
+
+  constructor(config: Config) {
+    this.socket = this.createNewSocket();
+    this.config = config;
```

Comment:
I kept most of the logic the same and used the `config` constructor argument to pass in the callbacks.